web/src/routes/CategoriesPage.tsx

```diff
@@ -50,12 +48,9 @@ const getBreadcrumb = (category: CategoryModel, cityName: string) => {
   })
 }
 
-type PropsType = CityRouteProps & RouteProps<typeof CATEGORIES_ROUTE>
-
-const CategoriesPage = ({ cityModel, match, location, languages }: PropsType): ReactElement => {
-  const previousPathname = useRef<string | null | undefined>(null)
-  const { cityCode, languageCode, categoryId } = match.params
-  const pathname = normalizePath(location.pathname)
+const CategoriesPage = ({ cityModel, pathname, languages, cityCode, languageCode }: CityRouteProps): ReactElement => {
+  const previousPathname = useRef<string | null>(null)
+  const categoryId = useParams()['*']
```

Comment:
Not really intuitive. What does the `*` stand for? The placeholder?

Reply:
It stands for a wildcard, i.e. matching any number of any path segments. See: https://reactrouter.com/docs/en/v6/upgrading/v5#note-on-route-path-patterns
This is necessary for allowing category paths of arbitrary length.